frontend/src/app/core/services/team.service.ts

```diff
@@ -4,33 +4,74 @@ import { INewTeam } from '@core/models/INewTeam';
 import { ITeam } from '@core/models/ITeam';
 import { IUpdateTeam } from '@core/models/IUpdateTeam';
 import { HttpInternalService } from '@core/services/http-internal.service';
-import { Observable } from 'rxjs';
+import { BehaviorSubject, Observable, Subject } from 'rxjs';
 
 @Injectable({
     providedIn: 'root',
 })
 export class TeamService {
     public routePrefix = '/team';
 
-    // eslint-disable-next-line no-empty-function
-    constructor(private httpService: HttpInternalService) {}
+    private emitTeamCreationSource = new Subject<ITeam>();
+
+    public teamCreationEmitted$ = this.emitTeamCreationSource.asObservable();
+
+    private emitTeamDeletionSource = new Subject<number>();
+
+    public teamDeletionEmitted$ = this.emitTeamDeletionSource.asObservable();
+
+    private currentTeamSource = new BehaviorSubject<number | undefined>(undefined);
+
+    public currentTeamEmitted$ = this.currentTeamSource.asObservable();
+
+    constructor(private httpService: HttpInternalService) {
+        this.getCurrentUserTeams()
+            .subscribe(teams => {
+                if (teams.length > 0) {
+                    this.emitCurrentTeamChange(teams[0].id);
+                }
+            });
+    }
+
+    public emitTeamCreation(newTeam: ITeam) {
+        this.emitTeamCreationSource.next(newTeam);
+    }
+
+    public emitTeamDeletion(deletedTeamId: number) {
+        this.emitTeamDeletionSource.next(deletedTeamId);
+    }
+
+    public emitCurrentTeamChange(currentTeamId: number) {
+        this.currentTeamSource.next(currentTeamId);
+    }
 
     public getCurrentUserTeams(): Observable<ITeam[]> {
         return this.httpService.getRequest<ITeam[]>(`${this.routePrefix}/user-teams`);
     }
 
-    public validatePageLink(teamId: number, pageLink: string) {
-        return this.httpService.getRequest<boolean>(`${this.routePrefix}/validatepagelink`, {
-            id: teamId,
-            pagelink: pageLink,
-        });
+    public getCurrentUserAdminTeams(): Observable<ITeam[]> {
+        return this.httpService.getRequest<ITeam[]>(`${this.routePrefix}/user-teams-admin`);
+    }
+
+    public validatePageLink(teamId: number | undefined, pageLink: string) {
```

Comment:
Why do we need undefined here?

Reply:
I use this method to validate link for both team creation or update. If team has id, input link is valid because it belongs to this team, but when the team is new, then link should be completely unique

Reply:
But you can just use an optional parameter, aren't you?

Reply:
Yes) I will fix it soon

Reply:
Fixed